sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptors/ResponseValidationInterceptor.java

```diff
@@ -0,0 +1,45 @@
+package com.azure.android.core.http.interceptors;
+
+import androidx.annotation.NonNull;
+
+import com.azure.android.core.util.function.BiConsumer;
+import com.azure.android.core.util.logging.ClientLogger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+public class ResponseValidationInterceptor implements Interceptor {
```

Comment:
Created only the interceptor class, no builder like in the [Java SDK](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/ResponseValidationPolicyBuilder.java) for now. Maybe @JonathanGiles or @alzimmermsft could shed some light into why the Java team decided to take the builder route.

Reply:
Let's redesign this a little bit. I see that in Java, the only functionality exposed / used is the builder's addOptionalEcho method, in order to verify a request header's value matches the same header's value in the response. We should be able to simplify this and move from a generic "response validator" to single, concrete "response header echo validator" or something (I'm sure we can come up with a better name). If there are other types of validations we want to add in the future, they can be added as different interceptors.

So my suggestion would be to make an interceptor with a constructor that can accept a list of header names, and whose `intercept` implementation matches the logic that `ResponseValidationPolicyBuilder.addOptionalEcho` currently performs in the Java SDK.

Reply:
So, summarizing, instead of having a class that can have a list of assertions to check the response against, we would be talking about an interceptor that validates that the values of a given list of headers are consistent between the request and the response?

Reply:
Right, because that is the only type of assertion we ever coded into the Java SDK. It seems like the Java SDK implemented it as a way to allow us to write any arbitrary type of assertions, but only one type of assertion has ever actually been created so far. I don't see the value in adding that complexity for Android.

Reply:
In the Java SDK, the `RequestValidationPolicy` class is public, doesn't this mean that it would also allow users to add their own assertions?

Reply:
I would assume so, but I don't personally feel like there's a compelling reason to expose that feature on mobile. However the buck stops with @adrianhall on that :)

Reply:
Agree with @bsiegel, we can keep it either as a package internal type or move it under implementation package. Storage specifically want to validate two headers `x-ms-client-id` and `x-ms-encryption-key-sha256`. If customers are interested in custom validations they can provide their own interceptors.